packages/next-app/libs/hooks/useLocalStorage.ts

```diff
@@ -0,0 +1,29 @@
+import { useState } from "react";
+
+export default function useLocalStorage<T>(key: string, initialValue: T) {
+	const [storedValue, setStoredValue] = useState<T>(() => {
+		if (typeof window === "undefined") {
+			return initialValue;
+		}
+		try {
+			const item = window.localStorage.getItem(key);
+			return item ? JSON.parse(item) : initialValue;
+		} catch (error) {
+			console.log(error);
+			return initialValue;
+		}
+	});
+	const setValue = (value: T | ((val: T) => T)) => {
```

Comment:
nice custom hook! i wonder if it's better to use `useEffect` to handle the initial `setStoredValue` part. As I can see here you have to do a bit of guarding due to server side rendering.

Server side rendering will never touch `useEffect`, so you can safely use `window` object in there.